agents/agents-ext/src/commonMain/kotlin/ai/koog/agents/ext/agent/AIAgentSubtaskExt.kt

```diff
@@ -0,0 +1,341 @@
+ * @param llmModel The optional large language model to be used during the subtask, if different from the default one.
+ * @param llmParams The configuration parameters for the large language model, such as temperature, etc.
+ * @param runMode The mode in which tools should be executed, either sequentially or in parallel.
+ * @param assistantResponseRepeatMax The maximum number of times the assistant response can repeat. Useful to control redundant outputs.
+ * @param defineTask A suspendable lambda defining the actual task logic, which takes the provided input and produces a task description.
+ * @return The result of the subtask execution, as an instance of type Output.
+ */
+@OptIn(InternalAgentToolsApi::class)
+public suspend inline fun <reified Input, reified Output> AIAgentFunctionalContext.subtask(
+    input: Input,
+    tools: List<Tool<*, *>>? = null,
+    llmModel: LLModel? = null,
+    llmParams: LLMParams? = null,
+    runMode: ToolCalls = ToolCalls.SEQUENTIAL,
+    assistantResponseRepeatMax: Int? = null,
+    defineTask: suspend AIAgentFunctionalContext.(input: Input) -> String
+): Output {
+    val finishTool = identityTool<Output>()
+
+    return subtask(input, tools, finishTool, llmModel, llmParams, runMode, assistantResponseRepeatMax, defineTask)
+}
+
+/**
+ * Executes a subtask within the AI agent's functional context. This method enables the use of tools to
+ * accomplish a specific task based on the input provided. It defines the task using an inline function,
+ * employs tools iteratively, and attempts to complete the subtask with a designated finishing tool.
+ *
+ * @param input The input data required to define and execute the subtask.
+ * @param tools An optional list of tools that can be utilized to accomplish the task, excluding the finishing tool.
+ * @param finishTool A mandatory tool that determines the final result of the subtask by producing and transforming output.
+ * @param llmModel An optional specific language learning model (LLM) to use for executing the subtask.
+ * @param llmParams Optional parameters for configuring the behavior of the LLM during subtask execution.
+ * @param runMode The mode in which tools should be executed, either sequentially or in parallel.
+ * @param assistantResponseRepeatMax The maximum number of feedback attempts allowed from the language model if the subtask is not completed.
+ * @param defineTask A suspendable function used to define the task based on the provided input.
+ * @return The transformed final result of executing the finishing tool to complete the subtask.
+ */
+@OptIn(InternalAgentToolsApi::class, DetachedPromptExecutorAPI::class, InternalAgentsApi::class)
+public suspend inline fun <reified Input, reified Output, reified OutputTransformed> AIAgentFunctionalContext.subtask(
+    input: Input,
+    tools: List<Tool<*, *>>? = null,
+    finishTool: Tool<Output, OutputTransformed>,
+    llmModel: LLModel? = null,
```

Comment:
Looks like `llmModel` and `llmParams` parameters are not used in this method's body.